src/core/elevation/qgsprofilesourceregistry.h

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************
+  qgsprofilesourceregistry.h
+  --------------------------------------
+  Date                 : April 2024
+  Copyright            : (C) 2024 by Germán Carrillo
+  Email                : german at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPROFILESOURCEREGISTRY_H
+#define QGSPROFILESOURCEREGISTRY_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+
+class QgsAbstractProfileSource;
+
+#include <QList>
+
+
+/**
+ * \ingroup core
+ * \brief Registry of profile sources used by QgsProfilePlotRenderer
+ *
+ * QgsProfileSourceRegistry is not usually directly created, but rather accessed through
+ * QgsApplication::profileSourceRegistry().
+ *
+ * \since QGIS 3.38
+ */
+class CORE_EXPORT QgsProfileSourceRegistry
```

Comment:
We could consider making this a QObject, with signals for sources added / removed, and automatically updating elevation plots when these signals are emitted.

It's not required, but would be a nice touch!

Reply:
Nice suggestion indeed, thanks. Unfortunately, we'll need to postpone this to an eventual followup, since it's been internally considered out of scope for this sprint.